platforms/core-execution/persistent-cache/src/main/java/org/gradle/cache/internal/scopes/DefaultCacheScopeMapping.java

```diff
@@ -32,11 +31,11 @@ public class DefaultCacheScopeMapping implements CacheScopeMapping {
     private static final Pattern CACHE_KEY_NAME_PATTERN = Pattern.compile("\\p{Alpha}+[-/.\\w]*");
 
     private final File globalCacheDir;
-    private final GradleVersion version;
+    private final String gradleVersion;
```

Comment:
:x: This is not a version now, but a generic string key. I'd rather keep this class bound to `GradleVersion` and move it to some other module that's shared in the right context. If we want to move it here, I suggest we call it `NamedCacheScopeMapping` and update it accordingly.